src/utils/xhr-utils.js

```diff
@@ -0,0 +1,624 @@
+ *     response to convert back to a regular Response.
+ * @param {string|undefined} responseType The original response type used to
+ *     initiate the XHR.
+ * @return {!FetchResponse|!Response} The deserialized regular response.
+ * @private
+ */
+export function fromStructuredCloneable(response, responseType) {
+  user().assert(isObject(response), 'Object expected: %s', response);
+
+  const isDocumentType = responseType == 'document';
+  if (typeof Response === 'function' && !isDocumentType) {
+    // Use native `Response` type if available for performance. If response
+    // type is `document`, we must fall back to `FetchResponse` polyfill
+    // because callers would then rely on the `responseXML` property being
+    // present, which is not supported by the Response type.
+    return new Response(response['body'], response['init']);
+  }
+
+  const lowercasedHeaders = map();
+  const data = {
+    status: 200,
+    statusText: 'OK',
+    responseText: (response['body'] ? String(response['body']) : ''),
+    /**
+     * @param {string} name
+     * @return {string}
+     */
+    getResponseHeader(name) {
+      return lowercasedHeaders[String(name).toLowerCase()] || null;
+    },
+  };
+
+  if (response['init']) {
+    const init = response['init'];
+    if (isArray(init.headers)) {
+      init.headers.forEach(entry => {
+        const headerName = entry[0];
+        const headerValue = entry[1];
+        lowercasedHeaders[String(headerName).toLowerCase()] =
+            String(headerValue);
+      });
+    }
+    if (init.status) {
+      data.status = parseInt(init.status, 10);
+    }
+    if (init.statusText) {
+      data.statusText = String(init.statusText);
+    }
+  }
+
+  if (isDocumentType) {
+    // TODO(prateekbh): remove this when integrating document fetcher.
+    data.responseXML =
+        new DOMParser().parseFromString(data.responseText, 'text/html');
+  }
+
+  return new FetchResponse(data);
+}
+
+/**
+ * Intercepts the XHR and proxies it through the viewer if necessary.
+ *
+ * XHRs are intercepted if all of the following are true:
+ * - The AMP doc is in single doc mode
+ * - The requested resource is not a 1p request.
+ * - The viewer has the `xhrInterceptor` capability
+ * - The Viewer is a trusted viewer or AMP is currently in developement mode
+ * - The AMP doc is opted-in for XHR interception (`<html>` tag has
+ *   `allow-xhr-interception` attribute)
+ *
+ * @param {!Window} win
+ * @param {?../service/ampdoc-impl.AmpDoc} ampdocSingle
+ * @param {string} input The URL of the XHR which may get intercepted.
+ * @param {!FetchInitDef} init The options of the XHR which may get
+ *     intercepted.
+ * @return {!Promise<!FetchResponse|!Response|undefined>}
+ *     A response returned by the interceptor if XHR is intercepted or
+ *     `Promise<undefined>` otherwise.
+ * @private
+ */
+export function getViewerInterceptResponse(win, ampdocSingle, input, init) {
```

Comment:
@jridgewell this is just a renamed function of [maybeInterceptViewer](https://github.com/ampproject/amphtml/blob/master/src/service/xhr-impl.js#L184)

with removal of `this....` variables

Reply:
@jridgewell this is just a renamed function of maybeInterceptViewer

with removal of this.... variables